src/components/forms/Election/index.tsx

```diff
@@ -1,55 +1,49 @@
 import Button from "@/components/buttons/Button";
 import ListingCard from "@/components/cards/ListingCard";
-import {CardInterface} from "@/utils/types/card";
+import LocationInput from "@/components/input/LocationInput";
+import { CardInterface } from "@/utils/types/card";
+import { Feature } from "@/utils/types/restaurant-finder";
+import { Group, NumberInput } from "@mantine/core";
+import axios from "axios";
+import { useCallback, useState } from "react";
+
+const Election = ({ theme }: { theme: string }) => {
+    const [restaurants, setRestaurants] = useState<Feature[]>([]);
+    const [isLoading, setIsLoading] = useState<boolean>(false);
+    const [searchRadius, setSearchRadius] = useState(5);
 
-const foodList: CardInterface[] = [
-    {
-        id: "1",
-        restaurantName: "KFC",
-        location: "Kuala Lumpur",
-        image: "https://raw.githubusercontent.com/mantinedev/mantine/master/.demo/images/bg-8.png",
-        cuisine: ["Fast Food"]
-    },
-    {
-        id: "2",
-        restaurantName: "McDonald's",
-        location: "Kuala Lumpur",
-    },
-    {
-        id: "3",
-        restaurantName: "Burger King Kongek Besar Jahanam",
-        image: "https://raw.githubusercontent.com/mantinedev/mantine/master/.demo/images/bg-8.png",
-        cuisine: ["King Kong", "Fast Food"]
-    },
-    {
-        id: "4",
-        restaurantName: "Subweiweiweiweiweiweiweiweiweiweiweiweiweiwei",
-        location: "Kuala Lumpur",
-        image: "https://raw.githubusercontent.com/mantinedev/mantine/master/.demo/images/bg-8.png",
-        cuisine: ["Sandwiches"]
-    },
-    {
-        id: "5",
-        restaurantName: "Texas Chicken",
-        location: "Kuala Lumpur",
-        image: "https://raw.githubusercontent.com/mantinedev/mantine/master/.demo/images/bg-8.png",
-        cuisine: ["Down south"]
-    }
-];
+    const fetchRestaurants = useCallback(() => {
+        if (isLoading === true) return;
 
-const Election = ({theme}: { theme: string }) => {
+        navigator.geolocation.getCurrentPosition(
+            async (pos) => {
+                setIsLoading(true);
+                const res = await axios("/api/restaurant-finder", {
+                    params: { lat: pos.coords.latitude, lng: pos.coords.longitude, radius: searchRadius },
+                });
+                setRestaurants(res.data);
+                setIsLoading(false);
+            },
+            (err) => {
+                console.log(err);
+                setIsLoading(false);
+            },
+            { enableHighAccuracy: true },
+        );
+    }, [isLoading, searchRadius]);
 
-    return <>
-        <LocationInput theme={theme}/>
-        <Group position="right" mt="md" pt={8}>
-            <Button color={theme}>
-                Start Voting!
-            </Button>
-        </Group>
-        <ListingCard theme={theme} cardList={foodList}/>
-    </>
+    return (
+        <>
+            <LocationInput theme={theme} />
```

Comment:
do u have some api that has the location name like Cheras, Selangor. populate the `useLocationStore` so that the input text is populated accordingly to what location the cards are generated with